orchestration/dagster_orchestration/hca_orchestration/solids/validate_egress.py

```diff
@@ -0,0 +1,92 @@
+import os
+from typing import Any
+
+from dagster import configured, solid, InputDefinition, String, DagsterType
+
+from hca_manage.manage import HcaManage, ProblemCount
+
+
+def problem_count_typecheck(_, value: Any) -> bool:
+    return isinstance(value, ProblemCount)
+
+
+DagsterProblemCount: DagsterType = DagsterType(
+    name="DagsterProblemCount",
+    type_check_fn=problem_count_typecheck,
+    description="A simple named tuple to represent the different types of issues "
+                "present from the post process validation.",
+)
+
+
+POST_VALIDATION_SETTINGS_SCHEMA = {
+    "gcp_env": String,
+    "dataset_name": String,
+}
+
+
+@solid(
+    required_resource_keys={"data_repo_client"},
+    config_schema={
+        **POST_VALIDATION_SETTINGS_SCHEMA,
+        "google_project_name": String,
+    }
+)
+def base_post_import_validate(context) -> DagsterProblemCount:
+    """
+    Checks if the target dataset has any rows with duplicate IDs or null file references.
+    """
+    validator = HcaManage(
+        environment=context.solid_config["gcp_env"],
+        project=context.solid_config["google_project_name"],
+        dataset=context.solid_config["dataset_name"],
+        data_repo_client=context.resources.data_repo_client)
+    return validator.check_for_all()
+
+
+# sets up default config settings to minimize pipeline boilerplate
+@configured(base_post_import_validate, {"dataset_name": String})
+def post_import_validate(config):
+    return {
+        'gcp_env': os.environ.get("HCA_GCP_ENV"),
+        'google_project_name': os.environ.get("HCA_GOOGLE_PROJECT"),
+        **config,
+    }
+
+
+@solid(
+    required_resource_keys={"slack"},
+    input_defs=[InputDefinition("validation_results", DagsterProblemCount)],
+    config_schema={
+        **POST_VALIDATION_SETTINGS_SCHEMA,
+        "channel": String,
+    }
+)
+def base_notify_slack_of_egress_validation_results(context, validation_results) -> str:
```

Comment:
❓It'd be useful to thread through and include the initiating argo workflow ID as we will very likely have concurrent imports running. Feel free to push back as out of scope and we can ticket as a small bite.

Reply:
yeah, this seems like it'd be better as a separate ticket.